apps/src/code-studio/components/ShareAllowedDialog.jsx

```diff
@@ -323,227 +323,270 @@ class ShareAllowedDialog extends React.Component {
+                    isLegacyShare={false}
+                  />
                 </div>
-                {this.state.showSendToPhone && (
-                  <div>
-                    <div style={styles.sendToPhoneContainer}>
-                      <div style={styles.sendToPhoneLeft}>
-                        <SendToPhone
-                          channelId={channelId}
-                          appType={appType}
-                          isLegacyShare={false}
-                        />
-                      </div>
-                      <div style={styles.sendToPhoneRight}>
-                        <label>{i18n.scanQRCode()}</label>
-                        <QRCode value={shareUrl + '?qr=true'} size={90} />
-                      </div>
-                    </div>
-                    <div style={{clear: 'both'}} />
-                  </div>
-                )}
-                {warningText && (
-                  <div style={styles.warningMessageContainer}>
-                    <span
-                      style={styles.thumbnailWarning}
-                      className="thumbnail-warning"
-                    >
-                      {warningText}
-                    </span>
-                  </div>
-                )}
-                <div style={{clear: 'both', marginTop: 40}}>
-                  {isDroplet && (
-                    <AdvancedShareOptions
-                      shareUrl={shareUrl}
-                      exportApp={exportApp}
-                      expanded={this.state.showAdvancedOptions}
-                      onExpand={this.showAdvancedOptions}
-                      channelId={channelId}
-                      embedOptions={embedOptions}
-                      appType={this.props.appType}
-                    />
-                  )}
+                <div style={styles.sendToPhoneRight}>
+                  <label>{i18n.scanQRCode()}</label>
+                  <QRCode value={shareUrl + '?qr=true'} size={90} />
                 </div>
               </div>
+              <div style={{clear: 'both'}} />
             </div>
           )}
+          {warningText && (
+            <div style={styles.warningMessageContainer}>
+              <span
+                style={styles.thumbnailWarning}
+                className="thumbnail-warning"
+              >
+                {warningText}
+              </span>
+            </div>
+          )}
+          <div style={{clear: 'both', marginTop: 40}}>
+            {isDroplet && (
+              <AdvancedShareOptions
+                shareUrl={shareUrl}
+                exportApp={exportApp}
+                expanded={this.state.showAdvancedOptions}
+                onExpand={this.showAdvancedOptions}
+                channelId={channelId}
+                embedOptions={embedOptions}
+                appType={this.props.appType}
+              />
+            )}
+          </div>
+        </div>
+      );
+    };
+
+    const renderInterfaceV2 = () => {
```

Comment:
Similar to Jim's comment, but thoughts on creating a separate component for this? And additionally, should we just handle the branching logic in `Lab2ShareDialog.tsx`? We can bypass rendering `ShareAllowedDialog` entirely then, since this share view is quite different and I'd expect it will continue to deviate from the existing share UI.

Reply:
You're right.  I had been concerned about supporting all of the variants of the legacy dialog (as can be seen in the Storybook [here](https://github.com/code-dot-org/code-dot-org/blob/5285a67de0eedb3d14b5bdf115c1b18c0a542619/apps/src/code-studio/components/ShareAllowedDialog.story.jsx)), though the more I looked, the more it seemed that probably only `isAbusive` was a likely candidate.  I opened an alternative approach in https://github.com/code-dot-org/code-dot-org/pull/57333 which is a modern functional React component, and doesn't yet support any of those variants, though it certainly can as we learn more.